### PR TITLE
Add user auth capability in hapi asap

### PR DESCRIPTION
**Pull request by ish-bindra**
https://ordermentum.atlassian.net/browse/C1-515

There's a need to have a combined authorization (Asap + User auth) in hapi services. Till now we have been using a stop-gap solution that:
- Does not provide an `authorization` wrapped object on the request but returns `asapClaims` for asap authentication
- Is a new plugin that's registered in a different plugin and is not a sustainable solution

This PR 
- adds capability for the `hapi-asap` plugin to register a user auth strategy on the server if it's provided the right options.
- It then registers a `default` strategy which is a combined authorization strategy. (Asap, User auth in order of verification)
- Returns an `authorization` wrapped object that has all the niceties of the auth driver (If service user id is used)

### CODE REVIEW

File: packages/hapi-asap/src/middleware.ts [hunk 1]
```diff
@@ -4,6 +4,9 @@ import {
   createAsapAuthenticator,
   AuthenticatorOptions,
 } from '@ordermentum/asap-core';
+import { getAdminAuth } from '@ordermentum/auth-driver';
+import { HapiAuthPluginOptions, createHapiAuthPlugin } from '@ordermentum/auth-middleware';
```
Comment (johndagostino) [inline packages/hapi-asap/src/middleware.ts:8]: @ish-bindra think this seems worthwhile directionally, but does 1. make a hard dependency on a closed source package on an open source package, 2. is very ordermentum specific

how do you suggest we could approach both those?

Comment (ish-bindra) [reply packages/hapi-asap/src/middleware.ts:8]: Hmm, tbh i didn't realise this was open source. 
Think the better approach would be to go the other way around and utilise asap strategy through the auth middleware.

**Comment by rrsmith** [comment]
@ish-bindra I don't think this PR remains now right? Adding internal packages to the open source one is a no go

**Comment by ish-bindra** [comment]
> @ish-bindra I don't think this PR remains now right? Adding internal packages to the open source one is a no go

Yup, wanted your thoughts on https://github.com/ordermentum/asap-javascript/pull/29#discussion_r1253941975

**Comment by rrsmith** [comment]
There are a couple of ways we can approach it:
* instead of bundling into a single auth strategy we could create a module for a prehandler that checks and rejects calls if no auth credentials are set that is used with hapis `try` auth strategy
* we just use the custmCheck already built into auth middleware to use to test the asap strategy - in fact we could still include it prepackaged ready to go in auth-middleware but under the hood it could be a very simple adaption of the existing method

I prefer the 2nd option now that I've double checked auth-middleware (Wasn't so keen on us having to rewrite a bunch)

**Pull request closed without merging**